regexp_test.go

```diff
@@ -853,6 +853,89 @@ func TestECMANamedGroup(t *testing.T) {
 	}
 }
 
+func TestECMAGroupNameUnicode(t *testing.T) {
+	t.Run("unicode-escape", func(t *testing.T) {
```

Comment:
since this is an ECMA-script only change can we get an inverse test (i.e. make sure this errors without the ECMAScript flag)?